system-report.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+#
+# This file is part of Canonical's TDX repository which includes tools
+# to setup and configure a confidential computing environment
+# based on Intel TDX technology.
+# See the LICENSE file in the repository for the license text.
+
+# Copyright 2024 Canonical Ltd.
+# SPDX-License-Identifier: GPL-3.0-only
+
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 3,
+# as published by the Free Software Foundation.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranties
+# of MERCHANTABILITY, SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU General Public License for more details.
+
+#
+# This script outputs relevant system information for
+# reporting TDX issues at:
+#
+# https://github.com/canonical/tdx/issues
+#
+
+print_section() {
+  if [ $# -ne 2 ]; then
+    >&2 echo "$0 <header> <command output>"
+    exit 1
+  fi
+  header=$1
+  cmdout=$2
+  printf "### ${header}\n\n"
+  printf "\`\`\`\n"
+  printf "${cmdout}"
+  printf "\n\`\`\`\n\n"
+}
+
+set_pkg_result_string() {
+  if [ $# -ne 1 ]; then
+    >&2 echo "$0 <package>"
+    exit 1
+  fi
+  package=$1
+  result=$( \
+  if dpkg -s ${package} &> /dev/null; then \
+    echo "Status: Installed"; \
+  else echo "Status: Not Installed"; \
+  fi)
+  result="$result\n$(apt info ${package} 2>/dev/null | grep -E 'Package|Version|APT-Sources')"
+}
+
+printf "If you are running this for reporting an issue on GitHub,\n"
+printf "copy all output between the markers below.\n\n"
+
+printf "<======== COPY BELOW HERE ========>\n\n"
+
+result=$(lsb_release -a)
+print_section "Operating system details" "${result}"
+
+result=$(uname -rvpio) # show everything but hostname
+print_section "Kernel version" "${result}"
+
+result=$(sudo dmesg | grep -i tdx)
+print_section "TDX kernel logs" "${result}"
+
+result=$( \
+if grep -q tdx /proc/cpuinfo; then \
```

Comment:
see if possible to dump relevant MSR values (see `test_host_tdx_hardware.py`) but it will need sudo ...

Reply:
I'll check

Reply:
Other quick note in case it causes confusion, I had a typo in the line above that caused this result to state that the CPU did not contain TDX support. I have since fixed the typo and edited the original PR description to include the updated output.

Reply:
@hector-cao I added a new section with the MSR values.